pkg/crypto/keys/publickey.go

```diff
@@ -406,3 +408,19 @@ func (p *PublicKey) UnmarshalYAML(unmarshal func(any) error) error {
 	}
 	return p.DecodeBytes(b)
 }
+
+func (p *PublicKey) Copy() *PublicKey {
+	if p == nil {
+		return nil
+	}
+	copied := *p
+	if p.X != nil {
+		copiedX := new(big.Int).Set(p.X)
+		copied.X = copiedX
+	}
+	if p.Y != nil {
+		copiedY := new(big.Int).Set(p.Y)
+		copied.Y = copiedY
+	}
+	return &copied
+}
```

Comment:
OK, I think we need @roman-khimov's opinion on this. Do we need such deep copies?

Reply:
Discussed, we don't need this for `PublicKey`, but need to add documentation about this behaviour.